### ISSUE TITLE
Weblinks issue

### ISSUE THREAD

**Issue by nicksavov** [OSTraining/OSMap#36]
See:
https://secure.helpscout.net/conversation/77236155/11709/

**Comment by andergmartins** [comment]
Right, I will check, run the migration tool twice would explain that. The v2.4 wasn't removing the Xmap data after migrating, but v3.0 removes. Maybe after install the v3 he saw the migration button again, and migrated the data again (since we don't touch current data without the user action) because the older version didn't wiped it.

But I will double check making some tests

**Comment by andergmartins** [comment]
Ok, I will check if we need to update the weblinks plugin, otherwise I will fix it

**Comment by htmgarcia** [comment]
Same reason here: https://secure.helpscout.net/conversation/78771060/11823/

The sitemap breaks when find a menu item type: weblinks
If I disable that menu item, the sitemap works fine.

**Comment by htmgarcia** [comment]
This is odd, I tested again in local. Is working fine.

Now a new possible issue, the weblinks crawled automatically are the ones that redirect to the custom weblink url, which IMO are not required in a XML sitemap; simply won't work.

**Comment by photodude** [comment]
I can't read your helpscout discussions, but is this issue related to the J3.4.1/OSMaps/weblinks issue that causes a SQL error? (I've tested against J3.4.2RTC which does appear to fix the SQL issue)
```
1146 Table '#__weblinks' doesn't exist SQL=SELECT COUNT(id) AS count_links FROM #__weblinks WHERE state = 1
```

**Comment by htmgarcia** [comment]
Hi @photodude 
Since Weblinks was removed from Joomla 3.4.1, that error is expected due that component doesn't exist in fresh Joomla installations.

Please go to Extensions > Plugin manager > search "OSMap - WebLinks Plugin" and disable

**Comment by photodude** [comment]
Thanks @htmgarcia I tried disabling the OSMap - WebLinks Plugin but we still got the SQL error. 
Didn't try uninstalling the plugin though... I applied the 3.4.2RTC fix as a core hack/patch to resolve that message. So it looks like it wont be an issue in 3.4.2.
